tests/bson-deserialize-typewrapped-error-001.phpt

```diff
@@ -0,0 +1,31 @@
+--TEST--
+BSON serialization: wrong data types for type mappings
+--FILE--
+<?php
+$tests = [
+	[ 'types' => 'string' ],
+	[ 'types' => [ 'Does not exist' => 'Foo' ] ],
+	[ 'types' => [ 'Binary' => 43 ] ],
+	[ 'types' => [ 'Decimal128' => 'EmptyDecimalWrapper' ] ],
+];
+
+class EmptyDecimalWrapper {}
+
+foreach ( $tests as $test )
+{
+	try
+	{
+		$bson = MongoDB\BSON\fromPHP( [ 'random' => 'stuff' ] );
+		$test = MongoDB\BSON\toPHP( $bson, $test );
+	}
+	catch ( MongoDB\Driver\Exception\InvalidArgumentException $e )
+	{
+		echo $e->getMessage(), "\n";
```

Comment:
Should you be verifying the exception class as well? I assume you don't have the `throws()` utility function found in PHPC, but that would help here.